src/report/user.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  ActionLogger,
+  Author,
+  IssueNode,
+  PullRequestNode,
+} from "../github/types";
+import { calculateEventsPerMonth, gatherValuesPerMonth } from "../util";
+import { PullRequestAnalytics } from "./pullRequests";
+import { MonthWithMatch } from "./types";
+
+export interface UserMetrics {
+  created: MonthWithMatch[];
+  closed: MonthWithMatch[];
+  reviewsPerMonth: MonthWithMatch[];
+  commentsPerReview: MonthWithMatch[];
+  participatedIssuesPerMonth: MonthWithMatch[];
+  nonOrgParticipatedIssuesPerMonth: MonthWithMatch[];
+}
+
+export class UserAnalytics {
+  constructor(
+    private readonly logger: ActionLogger,
+    repo: { owner: string; repo: string },
+    private readonly author: string,
+  ) {
+    logger.debug(
+      `Reporter has been configured for ${author}'s contribution in ${repo.owner}/${repo.repo}`,
+    );
+  }
+
+  private isAuthor(author: Author): boolean {
+    return (
+      author?.login?.toLocaleUpperCase() === this.author.toLocaleUpperCase()
+    );
+  }
+
+  public generateMetrics(
+    prList: PullRequestNode[],
+    issues: IssueNode[],
+  ): UserMetrics {
+    const authorship = this.generateMetricsForPullRequestsAuthorship(prList);
+    const reviews = this.generateMetricsForPullRequestsReviews(prList);
+    const issueMetrics = this.generateIssuesMetrics(issues);
+    return { ...authorship, ...reviews, ...issueMetrics };
```

Comment:
Are we just joining all issues, reviews and PRs?

Reply:
Yes, as the final object is 
```typescript
export interface UserMetrics {
  created: MonthWithMatch[];
  closed: MonthWithMatch[];
  reviewsPerMonth: MonthWithMatch[];
  commentsPerReview: MonthWithMatch[];
  participatedIssuesPerMonth: MonthWithMatch[];
  nonOrgParticipatedIssuesPerMonth: MonthWithMatch[];
}
``` and each method return two values of the object

Reply:
Ah, okay, I see